src/util/sandbox.cpp

```diff
@@ -385,66 +401,76 @@ QString Sandbox::migrateOldSettings() {
     // even if they are installing Mixxx >= 2.3.0 without having installed an old version of Mixxx.
     QString title = QObject::tr("Upgrading old Mixxx settings");
-    QString oldPath = homePath + QLatin1String("/Library/Application Support/Mixxx");
     QMessageBox::information(nullptr,
-            title,
-            QObject::tr(
-                    "Due to Mac sandboxing, Mixxx needs your permission to "
-                    "access your music library "
-                    "and settings from Mixxx versions before 2.3.0. After "
-                    "clicking OK, you will see a file picker. "
-                    "To give Mixxx permission, press the Ok button in the file "
-                    "picker."
-                    "\n\n"
-                    "If you do not want to grant Mixxx access click Cancel on "
-                    "the file picker."));
+        title,
+        QObject::tr(
+            "Existing Mixxx settings directory must be moved to new location."
+            "\n\n"
+            "Due to MacOS sandboxing, Mixxx needs your permission to access your "
+            "music library and move settings to the sandbox from Mixxx versions "
+            "before 2.3.0. After clicking OK, you will see standars file open dialog. "
+            "\n\n"
+            "Select Open button in the file open dialog to grant Mixxx access to old files."
+            "\n\n"
+            "If you do not want to grant Mixxx access click Cancel on the file picker."
+        )
+    );
     QString result = QFileDialog::getExistingDirectory(
-            nullptr,
-            title,
-            oldPath);
-    if (result != oldPath) {
-        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate "
-                   "old settings from"
-                << oldPath << "User selected" << result;
+        nullptr,
+        title,
+        legacySettingsPath);
+    if (result != legacySettingsPath) {
+        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate old settings from"
+                << legacySettingsPath << "User selected" << result;
         return sandboxedPath;
     }
 
-    // Sandbox::askForAccess cannot be used here because it depends on settings being
-    // initialized. There is no need to store the bookmark anyway because this is a
-    // one time process.
-#ifdef __APPLE__
+    // Check if Mixxx process is sandboxed
+    QString sandboxContainerId = QProcessEnvironment::systemEnvironment().value("APP_SANDBOX_CONTAINER_ID", "");
+
     CFURLRef url = CFURLCreateWithFileSystemPath(
-            kCFAllocatorDefault, QStringToCFString(oldPath), kCFURLPOSIXPathStyle, true);
+            kCFAllocatorDefault, QStringToCFString(legacySettingsPath), kCFURLPOSIXPathStyle, true);
     if (url) {
         CFErrorRef error = NULL;
-        CFDataRef bookmark = CFURLCreateBookmarkData(
+        if (sandboxContainerId != "") {
```

Comment:
```suggestion
        if (!sandboxContainerId.isEmpty()) {
```

> Qt makes a distinction between null strings and empty strings for historical reasons. For most applications, what matters is whether or not a string contains any data, and this can be determined using the isEmpty() function.

https://doc.qt.io/qt-5/qstring.html#isEmpty